src/controllers/auth.controller.ts

```diff
@@ -40,13 +46,31 @@ export class AuthController {
     try {
       const token = await this.sessionService.signIn({ email, password });
       return { token };
-    } catch (error) {
-      switch (error?.message) {
+    } catch (error: any) {
+      const errorMessage = error?.message;
+      switch (errorMessage) {
```

Comment:
maybe we could refactor this `switch` to an object, like:

```
const errorMessages = {
whatever is in --> Errors.MISSING_PARAMS: new BadRequestError(
          whatever is in --> Errors.MISSING_PARAMS
        ),
default: new UnauthorizedError('not authorized')
}

throw errorMessages[errorMessage] || errorMessages['default'];
```

if interested : 

https://betterprogramming.pub/dont-use-if-else-and-switch-in-javascript-use-object-literals-c54578566ba0

https://www.youtube.com/watch?v=0NlsJuwFsrQ&ab_channel=midudev

Reply:
Issue created with this topic!
https://github.com/rootstrap/node-ts-api-base/issues/599